services/src/main/java/org/keycloak/protocol/oidc/grants/ciba/channel/AuthenticationRequest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ *
+ *  * Copyright 2021  Red Hat, Inc. and/or its affiliates
+ *  * and other contributors as indicated by the @author tags.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+package org.keycloak.protocol.oidc.grants.ciba.channel;
+
+import javax.crypto.SecretKey;
+import java.io.UnsupportedEncodingException;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.crypto.Algorithm;
+import org.keycloak.crypto.KeyUse;
+import org.keycloak.crypto.SignatureProvider;
+import org.keycloak.crypto.SignatureSignerContext;
+import org.keycloak.jose.jwe.JWEException;
+import org.keycloak.jose.jws.JWSBuilder;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.protocol.oidc.grants.ciba.CibaGrantType;
+import org.keycloak.representations.IDToken;
+import org.keycloak.representations.JsonWebToken;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.keycloak.services.Urls;
+import org.keycloak.util.TokenUtil;
+
+/**
+ * <p>Represents an authentication request sent by a consumption device.
+ *
+ * <p>A authentication request can be serialized to a JWE so that they can be exchanged with authentication devices to communicate
+ * and authorize the authentication request made by consumption devices.
+ */
+public class AuthenticationRequest extends JsonWebToken {
```

Comment:
Minor: I vote to rename this class to CIBAAuthenticationRequest. We already have class "AuthenticationRequest" used by brokering, so it may help to avoid some confusions. In theory package name differentiation should be sufficient, but still sometimes the confusion can easily happen for the duplicated class names. But maybe it is just me... :-)

Reply:
I don't mind. But I would prefer the way it is.

Reply:
We can leave to @tnorimat whatever he prefers. I am also fine with either way to keep or rename.

Reply:
@mposolda @pedroigor My vote is to rename it to `CIBAAuthenticationRequest`.

Reply:
@mposolda I've renamed `CIBAAuthenticationRequest` to (https://github.com/keycloak/keycloak/pull/7679/commits/4d1e3475321a4d259cb9b9825fd3035e41a12aed).